src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommand.cs

```diff
@@ -0,0 +1,162 @@
+                    var sources = ParseArgumentWithCommas(source.Value());
+                    
+                    var packageSources = GetPackageSources(settings, sources, config);
+
+                    var frameworks = ParseArgumentWithCommas(framework.Value());
+                    
+                    var packageRefArgs = new ListPackageArgs(
+                        path.Value,
+                        packageSources,
+                        frameworks,
+                        includeOutdated.HasValue(),
+                        includeDeprecated.HasValue(),
+                        includeTransitive.HasValue(),
+                        prerelease.HasValue(),
+                        highestPatch.HasValue(),
+                        highestMinor.HasValue(),
+                        logger,
+                        CancellationToken.None
+                    );
+
+                    var listPackageCommandRunner = getCommandRunner();
+                    await listPackageCommandRunner.ExecuteCommandAsync(packageRefArgs);
+                    return 0;
+                });
+            });
+        }
+
+        private static ISettings ProcessConfigFile(string configFile, string currentDirectory)
+        {
+            if (string.IsNullOrEmpty(configFile))
+            {
+                return Settings.LoadDefaultSettings(currentDirectory);
+            }
+            else
+            {
+                var configFileFullPath = Path.GetFullPath(configFile);
+                var directory = Path.GetDirectoryName(configFileFullPath);
+                var configFileName = Path.GetFileName(configFileFullPath);
+                return Settings.LoadDefaultSettings(
+                    directory,
+                    configFileName,
+                    machineWideSettings: new XPlatMachineWideSetting());
+            }
+
+        }
+
+        private static List<PackageSource> GetPackageSources(ISettings settings, List<string> sources, CommandOption config)
+        {
+            
+            var sourceProvider = new PackageSourceProvider(settings);
+            var availableSources = sourceProvider.LoadPackageSources().Where(source => source.IsEnabled).ToList();
+            var packageSources = new List<PackageSource>();
+            foreach (var source in sources)
+            {
+                packageSources.Add(PackageSourceProviderExtensions.ResolveSource(availableSources, source));
+            }
+
+            if (packageSources.Count == 0 || config.HasValue())
+            {
+                packageSources.AddRange(availableSources);
+            }
+
+            return packageSources;
+        }
+
+        private static List<string> ParseArgumentWithCommas(string argument)
+        {
+            if (argument == null)
+            {
+                return new List<string>();
+            }
+            var argumentList = argument.Split(',').ToList();
```

Comment:
I'm not sure what you mean

Reply:
you could do something like:

```
var argumentList = argument.Split(',').Select(a => a.Trim()).Where(a => string.IsNullOrEmpty(a)).ToList(); 
```

This will make sure there are no leading or trailing whitespace on any argument and empty argument will be ignored... basically parsing this:

```
"one,      two    , , three   "
```

to this:

```
{ "one", "two", three" }
```

instead of

```
{ "one", "      two   ", " ", "three   " }
```

Reply:
Oh ok